features/location/impl/src/main/kotlin/io/element/android/features/location/impl/send/SendLocationView.kt

```diff
@@ -0,0 +1,254 @@
+            bottomSheetState = rememberStandardBottomSheetState(initialValue = SheetValue.Expanded),
+        ),
+        sheetDragHandle = {},
+        sheetSwipeEnabled = false,
+        topBar = {
+            TopAppBar(
+                title = {
+                    Text(
+                        text = stringResource(CommonStrings.screen_share_location_title),
+                        style = ElementTheme.typography.aliasScreenTitle,
+                    )
+                },
+                navigationIcon = {
+                    BackButton(onClick = onBackPressed)
+                },
+            )
+        },
+    ) {
+        Box(
+            modifier = Modifier
+                .padding(it)
+                .consumeWindowInsets(it),
+            contentAlignment = Alignment.Center
+        ) {
+            MapboxMap(
+                styleUri = rememberTileServerUrl(),
+                modifier = Modifier.fillMaxSize(),
+                cameraPositionState = cameraPositionState,
+                uiSettings = MapDefaults.uiSettings,
+                symbolManagerSettings = MapDefaults.symbolManagerSettings,
+                locationSettings = MapDefaults.locationSettings.copy(
+                    locationEnabled = state.hasLocationPermission,
+                ),
+            )
+            Icon(
+                resourceId = DesignSystemR.drawable.pin,
+                contentDescription = null,
+                tint = Color.Unspecified,
+                modifier = Modifier.align { size, space, _ ->
+                    IntOffset(
+                        x = (space.width - size.width) / 2,
+                        y = (space.height / 2) - size.height,
+                    )
+                }
+            )
+            FloatingActionButton(
+                onClick = { state.eventSink(SendLocationEvents.SwitchToMyLocationMode) },
+                modifier = Modifier
+                    .align(Alignment.BottomEnd)
+                    .padding(end = 16.dp, bottom = 72.dp),
+            ) {
+                when (state.mode) {
+                    SendLocationState.Mode.PinLocation -> Icon(imageVector = Icons.Default.LocationSearching, contentDescription = null)
+                    SendLocationState.Mode.SenderLocation -> Icon(imageVector = Icons.Default.MyLocation, contentDescription = null)
+                }
+            }
+        }
+    }
+}
+
+@DayNightPreviews
+@Composable
+fun SendLocationViewPreview(
+    @PreviewParameter(SendLocationStateProvider::class) state: SendLocationState
+) = ElementPreview {
+    SendLocationView(
+        state = state,
+        onBackPressed = {},
+    )
+}
+
+@Composable
+private fun PermissionRationaleDialog(
+    onContinue: () -> Unit,
+    onDismiss: () -> Unit,
+    appName: String,
+) {
+    ConfirmationDialog(
+        content = stringResource(CommonStrings.error_missing_location_rationale_android, appName),
+        onSubmitClicked = { onContinue() },
```

Comment:
I think this could just be `onSubmitClicked = onContinue`, same for `onDismiss`.